### ISSUE TITLE
Bug: Не обновляются остатки по товарам после перехода на новую версию

### ISSUE THREAD

**Issue by evgrezanov** [wpcraft-ru/wooms#544]
**Как есть**
После обновления плагина до последней версии перестали синхронизироватся остатки по складам,  в Журнале нет сообщения об ошибках
в списке задач wooms_products_walker неудачно (рис1),  в логах сервера были ошибки
```
[Thu Feb 08 18:34:56.312542 2024] [fcgid:warn] [pid 13364:tid 140430272300800] [client 31.31.196.217:43274] mod_fcgid: read data timeout in 300 seconds
[Thu Feb 08 18:34:56.312624 2024] [core:error] [pid 13364:tid 140430272300800] [client 31.31.196.217:43274] End of script output before headers: admin-ajax.php
```
Отписал в саппорт - говорят надо VPS (рис2)

**Screenshots**
![image](https://github.com/wpcraft-ru/wooms/assets/5329288/7fb3c6fe-4dde-48e5-a89f-03c0fb143556)

![image](https://github.com/wpcraft-ru/wooms/assets/5329288/b55ce806-6993-460d-a523-5343634813fc)

**Дополнительные детали**

### wp-core ###

version: 6.4.3
site_language: ru_RU
user_language: ru_RU
timezone: +03:00
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 1
blog_public: 1
default_comment_status: open
environment_type: production
user_count: 28265
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai
wordpress_size: 311,62 МБ (326760459 bytes)
uploads_path: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/uploads
uploads_size: 27,25 ГБ (29254921678 bytes)
themes_path: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/themes
themes_size: 63,78 МБ (66879540 bytes)
plugins_path: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/plugins
plugins_size: 138,78 МБ (145516582 bytes)
database_size: 1,43 ГБ (1539774998 bytes)
total_size: 29,18 ГБ (31333853257 bytes)

### wp-active-theme ###

name: Electro (electro)
version: 3.3.15
author: MadrasThemes
author_website: https://madrasthemes.com/
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, automatic-feed-links, html5, custom-background, site-logo, woocommerce, wc-product-gallery-zoom, wc-product-gallery-lightbox, wc-product-gallery-slider, title-tag, post-formats, custom-logo, menus, widgets
theme_path: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/themes/electro
auto_update: Выключено

### wp-themes-inactive (4) ###

Twenty Nineteen: version: 1.4, author: Команда WordPress (latest version: 2.7), Автоматические обновления отключены
Twenty Twenty-Four: version: 1.0, author: Команда WordPress, Автоматические обновления отключены
Twenty Twenty-Three: version: 1.3, author: Команда WordPress, Автоматические обновления отключены
Twenty Twenty-Two: version: 1.6, author: Команда WordPress, Автоматические обновления отключены

### wp-plugins-active (15) ###

Classic Editor: version: 1.6.3, author: WordPress Contributors, Автоматические обновления отключены
Electro Extensions: version: 3.3.15, author: MadrasThemes, Автоматические обновления отключены
Min Max Quantity & Step Control for WooCommerce: version: 5.9, author: CodeAstrology Team (latest version: 6.2), Автоматические обновления отключены
Redux Framework: version: 4.4.11, author: Team Redux, Автоматические обновления отключены
Russian Post and EMS for WooCommerce: version: 1.3.8, author: Artem Komarov, Автоматические обновления включены
Saphali Woocommerce Lite: version: 1.9.3, author: Saphali, Автоматические обновления отключены
Slider Revolution: version: 6.6.19, author: ThemePunch, Автоматические обновления отключены
VDZ Yandex Metrika Plugin: version: 1.6.2, author: VadimZ, Автоматические обновления отключены
WooCommerce: version: 8.5.2, author: Automattic, Автоматические обновления отключены
WooMS: version: 9.10, author: WPCraft, Автоматические обновления отключены
WP All Import - WooCommerce Import Add-On Pro: version: 4.0.0, author: Soflyy, Автоматические обновления отключены
WP All Import Pro: version: 4.8.5, author: Soflyy, Автоматические обновления включены
WPBakery Page Builder: version: 7.3, author: Michael M - WPBakery.com (latest version: 7.4), Автоматические обновления отключены
YITH WooCommerce Wishlist: version: 3.27.0, author: YITH (latest version: 3.29.0), Автоматические обновления отключены
Yoast SEO: version: 21.7, author: Team Yoast (latest version: 22.0), Автоматические обновления отключены

### wp-plugins-inactive (1) ###

YITH WooCommerce Compare: version: 2.34.0, author: YITH (latest version: 2.35.0), Автоматические обновления отключены

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1690
imagemagick_version: ImageMagick 6.9.10-86 Q16 x86_64 2021-10-18 https://imagemagick.org
imagick_version: 3.5.1
file_uploads: 1
post_max_size: 256M
upload_max_filesize: 256M
max_effective_size: 256 МБ
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 251 ГБ
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 1536
	imagick::RESOURCETYPE_MAP: 251 ГБ
	imagick::RESOURCETYPE_MEMORY: 126 ГБ
	imagick::RESOURCETYPE_THREAD: 1
	imagick::RESOURCETYPE_TIME: 9.2233720368548E+18
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, ART, ARW, AVI, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, H, HALD, HDR, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBP, WMF, WMV, WMZ, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, WebP, BMP
ghostscript_version: 9.25

### wp-server ###

server_architecture: Linux 3.10.0-1160.105.1.el7.x86_64 x86_64
httpd_software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_fcgid/2.3.9 Phusion_Passenger/5.3.7
php_version: 8.0.30 64bit
php_sapi: cgi-fcgi
max_input_variables: 100000
time_limit: 300
memory_limit: 1024M
max_input_time: -1
upload_max_filesize: 256M
php_post_max_size: 256M
curl_version: 7.69.1 OpenSSL/1.0.2k-fips
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: true
current: 2024-02-09T07:07:10+00:00
utc-time: Friday, 09-Feb-24 07:07:10 UTC
server-time: 2024-02-09T10:07:06+03:00

### wp-database ###

extension: mysqli
server_version: 5.7.27-30
client_version: mysqlnd 8.0.30
max_allowed_packet: 33554432
max_connections: 2000

### wp-constants ###

WP_HOME: https://xn--80adbmlenjnkb2bkd6m.xn--p1ai
WP_SITEURL: undefined
WP_CONTENT_DIR: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content
WP_PLUGIN_DIR: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 1024M
WP_DEBUG: true
WP_DEBUG_DISPLAY: false
WP_DEBUG_LOG: true
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Не определено
WP_DEVELOPMENT_MODE: undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
0: Writable

### redux-framework ###

version: 4.4.11
installation: plugin
data directory: /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/plugins/redux-framework/redux-core/
browser: 
	Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
	Browser: Chrome
	Version: 121.0.0.0
	Platform: Apple

### redux-instance-electro_options ###

opt_name: electro_options
global_variable: electro_options
dev_mode: false
ajax_save: true
page_slug: theme_options
page_permissions: manage_options
menu_type: menu
page_parent: themes.php
compiler: true
output: true
output_tag: true
templates_path: undefined
extensions: 
	Accordion: 4.3.16
	Custom Fonts: 4.4.2
	Customizer: 4.4.11
	Datetime: 4.3.15
	Google Maps: 4.4.0
	Icon Select: 4.4.2
	Import Export: 4.0.0
	Js Button: 4.3.16
	Metaboxes: 4.2.0
	Multi Media: 4.4.1
	Options Object: 4.0.0
	Repeater: 4.3.13
	Search: 3.4.5
	Shortcodes: 4.3.6
	Social Profiles: 4.3.17
	Tabbed: 4.4.8
	Taxonomy: 4.4.6
	Users: 4.4.1
	Widget Areas: 4.3.20

### wooms-plugin-debug ###

Wooms Version: 9.10 ✔️
Woocommerce: 8.5.2 ✔️

**Бюджет**
@uptimizt стоит ли тут в оптимизацию играть и что-то смотреть в коде или реально может не хватать ресурсов хостинга?

В пачке ставил по 10-5 товаров, высего в магазине 5700 товаров простых,  1 склад

**Comment by evgrezanov** [comment]
upd
в debug.log еще такая ошибка вылезла @uptimizt 
```
[09-Feb-2024 10:12:31 UTC] PHP Warning:  Attempt to read property "term_id" on bool in /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/plugins/woocommerce/includes/wc-term-functions.php on line 452
[09-Feb-2024 10:12:33 UTC] WordPress database error Duplicate entry '1105608-793' for key 'PRIMARY' for query INSERT INTO wp_term_relationships (object_id, term_taxonomy_id)
					SELECT DISTINCT posts.ID, '793' FROM wp_posts posts
					LEFT JOIN
						(
							SELECT object_id FROM wp_term_relationships term_relationships
							LEFT JOIN wp_term_taxonomy term_taxonomy ON term_relationships.term_taxonomy_id = term_taxonomy.term_taxonomy_id
							WHERE term_taxonomy.taxonomy = 'product_cat'
						) AS tax_query
					ON posts.ID = tax_query.object_id
					WHERE posts.post_type = 'product'
					AND tax_query.object_id IS NULL made by do_action_ref_array('action_scheduler_run_queue'), WP_Hook->do_action, WP_Hook->apply_filters, ActionScheduler_QueueRunner->run, ActionScheduler_QueueRunner->do_batch, ActionScheduler_Abstract_QueueRunner->process_action, ActionScheduler_Action->execute, do_action_ref_array('wc_schedule_update_product_default_cat'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Internal\AssignDefaultCategory->maybe_assign_default_product_cat
[09-Feb-2024 10:12:33 UTC] PHP Warning:  Attempt to read property "term_id" on bool in /var/www/u0847658/data/www/xn--80adbmlenjnkb2bkd6m.xn--p1ai/wp-content/plugins/woocommerce/includes/wc-term-functions.php on line 452
```

**Comment by artikus11** [comment]
@evgrezanov есть там баг https://github.com/wpcraft-ru/wooms/issues/524#issuecomment-1860552168 Анатолий обещал в новом релизе поправить

**Comment by evgrezanov** [comment]
> @evgrezanov есть там баг [#524 (comment)](https://github.com/wpcraft-ru/wooms/issues/524#issuecomment-1860552168) Анатолий обещал в новом релизе поправить

ага решили,  спасибо
